cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/batch/BaseRDDCollection.java

```diff
@@ -113,7 +115,7 @@ protected BaseRDDCollection(JavaSparkExecutionContext sec, FunctionCache.Factory
   }
 
   @SuppressWarnings("unchecked")
-  @Override
+  @Nullable
```

Comment:
I don't really understand why this is now nullable. An RDDCollection should always be backed by an RDD. Wouldn't a null here also cause NullPointerExceptions the other places that it's called?

Reply:
I agree. Generally IMHO it should work pretty similar for all the getViewType() (RDD/DataSet/SQLDataSet). If it does not exists, just create/pull it.

Reply:
See answer in https://github.com/cdapio/cdap/pull/13343#discussion_r621668993

Reply:
But getUnderlying will do a pull, so it will never return null, will it? That's what I see in RDDCollection

Reply:
Good point. The RDD is not nullable as a class param, however it will never be null in practice.

Reply:
This is no longer used.